src/Aspire.Dashboard/Protos/Partials.cs

```diff
@@ -22,7 +22,7 @@ public ResourceViewModel ToViewModel()
             DisplayName = ValidateNotNull(DisplayName),
             Uid = ValidateNotNull(Uid),
             CreationTimeStamp = CreatedAt.ToDateTime(),
-            Properties = Properties.ToFrozenDictionary(p => ValidateNotNull(p.Name), p => ValidateNotNull(p.Value), StringComparers.ResourcePropertyName),
+            Properties = Properties.ToFrozenDictionary(property => ValidateNotNull(property.Name), property => ValidateNotNull(property.Value), StringComparers.ResourcePropertyName),
```

Comment:
nit: I'm not a fan of using frozen dictionary here. A resource will usually live a long time, but it could be updated frequently for state updates.

I've always keep frozen dicitonary to things that live the lifetime of an app.

Reply:
The idea here is that the view model is immutable. Changes produce new instances.

If we change the design to have these be mutable then we can change this too.

Reply:
Choosing something that is readonly is fine, e.g. ImmutableDictionary or IReadOnlyDictionary.

Frozen is about performance, and we haven't tested its impact. I really doubt it would make a difference either way though.

Reply:
In which ways is it about performance?

I chose it for after the guarantee of immutability. `IReadOnlyDictionary` only hints that the consumer cannot change it, though you don't know if the data changes underneath you. This makes things like enumeration risky. `Frozen` is a guarantee that the data is as-is forever.

`ImmutableDictionary` unfortunately squatted the better name. If we could turn back the clock, `FrozenDictionary` would have been called `ImmutableDictionary` and `ImmutableDictionary` would have been called `PersistentDictionary` (a la https://en.wikipedia.org/wiki/Persistent_data_structure).

We use `ImmutableDictionary` extensively in CPS and it's a source of plenty of performance woes. I would not recommend using it except in very specific circumstances (huge tree with minor mutations).

Reply:
Frozen dictionary is much slower to create - https://startdebugging.net/2023/08/net-8-performance-dictionary-vs-frozendictionary/

That's why it's typically used for dictionaries that last forever.

> IReadOnlyDictionary only hints that the consumer cannot change it, though you don't know if the data changes underneath you. 

I'm not worried about that. We control the type.

Reply:
That was an interesting read, though the benchmarks are outdated. The `optimizeForReading` was removed. Some day I'll try creating updated benchmarks for my own edification. For now I'll leave this as-is as I don't see a real issue here. I like the immutability guarantee, and any overhead is tiny and dwarfed by the kind of optimisation proposed in #1348, or the delay introduced due to not having OTEL retries.